src/playback_observer/types.ts

```diff
@@ -236,3 +238,87 @@ export interface IReadOnlyPlaybackObserver<TObservationType> {
   ): IReadOnlyPlaybackObserver<TDest>;
 }
+
+export interface ICorePlaybackObservation extends IPeriodStreamPlaybackObservation {
+  rebuffering: IRebufferingStatus | null;
+  freezing: IFreezingStatus | null;
+  bufferGap: number | undefined;
+}
+
+/** Regular playback information needed by the AdaptationStream. */
+export interface IAdaptationStreamPlaybackObservation {
+  /**
+   * Information on whether the media element was paused at the time of the
+   * Observation.
+   */
+  paused: IPausedPlaybackObservation;
+  /**
+   * Information on the current media position in seconds at the time of the
+   * Observation.
+   */
+  position: ObservationPosition;
+  /** `duration` property of the HTMLMediaElement. */
+  duration: number;
+  /** `readyState` property of the HTMLMediaElement. */
+  readyState: number;
+  /** Target playback rate at which we want to play the content. */
+  speed: number;
+  /** Theoretical maximum position on the content that can currently be played. */
+  maximumPosition: number;
+  /**
+   * For the current SegmentSink, difference in seconds between the next position
+   * where no segment data is available and the current position.
+   */
+  bufferGap: number;
+  /**
+   * Ranges of buffered data for the adaptation media type.
+   * `null` if no buffer exists for the media type.
+   */
+  buffered: IRange[] | null;
+}
+
+export interface IPeriodStreamPlaybackObservation {
+  /**
+   * Information on whether the media element was paused at the time of the
+   * Observation.
+   */
+  paused: IPausedPlaybackObservation;
+  /**
+   * Information on the current media position in seconds at the time of the
+   * Observation.
+   */
+  position: ObservationPosition;
+  /** `duration` property of the HTMLMediaElement. */
+  duration: number;
+  /** `readyState` property of the HTMLMediaElement. */
+  readyState: number;
+  /** Target playback rate at which we want to play the content. */
+  speed: number;
+  /** Theoretical maximum position on the content that can currently be played. */
+  maximumPosition: number;
+  /**
+   * Ranges of buffered data per type of media.
+   * `null` if no buffer exists for that type of media.
+   */
+  buffered: Record<ITrackType, IRange[] | null>;
+}
+
+/** Pause-related information linked to an emitted Playback observation. */
+export interface IPausedPlaybackObservation {
+  /**
+   * Known paused state at the time the Observation was emitted.
+   *
+   * `true` indicating that the HTMLMediaElement was in a paused state.
+   *
+   * Note that it might have changed since. If you want truly precize
+   * information, you should recuperate it from the HTMLMediaElement directly
+   * through another mean.
+   */
+  last: boolean;
+  /**
+   * Actually wanted paused state not yet reached.
+   * This might for example be set to `false` when the content is currently
+   * loading (and thus paused) but with autoPlay enabled.
+   */
+  pending: boolean | undefined;
+}
```

Comment:
To me the `PlaybackObserver` has no concept of what the `AdaptationStream` or `RepresentationStream` is.

For "core" I'm ok with.

This is mainly a naming issue for me.